pkg/server/controller_test.go

```diff
@@ -19,14 +21,52 @@ import (
 )
 
 const assetAtCompressionThreshold, assetLtCompressionThreshold = "AssetAtCompressionThreshold", "AssetLTCompressionThreshold"
+const tlsCert = 
+`-----BEGIN CERTIFICATE-----
+MIICpzCCAi6gAwIBAgIUVyJRQWSwWAdra4ndkDYGQ36wnBUwCgYIKoZIzj0EAwIw
+gYoxCzAJBgNVBAYTAlVTMQswCQYDVQQIDAJEQzELMAkGA1UEBwwCREMxFTATBgNV
+BAoMDHB5cm9zY29wZS5pbzEVMBMGA1UECwwMcHlyb3Njb3BlLmlvMRIwEAYDVQQD
+DAlsb2NhbGhvc3QxHzAdBgkqhkiG9w0BCQEWEGRldkBweXJvc2NvcGUuaW8wHhcN
+MjEwOTIwMTMwMDI0WhcNMzEwOTE4MTMwMDI0WjCBijELMAkGA1UEBhMCVVMxCzAJ
+BgNVBAgMAkRDMQswCQYDVQQHDAJEQzEVMBMGA1UECgwMcHlyb3Njb3BlLmlvMRUw
+EwYDVQQLDAxweXJvc2NvcGUuaW8xEjAQBgNVBAMMCWxvY2FsaG9zdDEfMB0GCSqG
+SIb3DQEJARYQZGV2QHB5cm9zY29wZS5pbzB2MBAGByqGSM49AgEGBSuBBAAiA2IA
+BBiXAzwxT5591cwYgExG5WTO4LkNua9fexZ595sHgoTs+l13hlmJxdFDaVaasf7W
+S/A+8hQO8CLEvAHVNPhXiSE7yQyLRc7kZNxdKXVFH/6RyErWL32XwD/kOKhtzr9T
+CqNTMFEwHQYDVR0OBBYEFB8iItOC7tw8Vt4WX8AWIWoTTmTlMB8GA1UdIwQYMBaA
+FB8iItOC7tw8Vt4WX8AWIWoTTmTlMA8GA1UdEwEB/wQFMAMBAf8wCgYIKoZIzj0E
+AwIDZwAwZAIwQ6kqjamnb73dM1ARyA1VfE0ZMKHPP/bX7t1GnjT9bQI5YtsL6txT
+Tsw7f3UVBQ9YAjBEy/MSnZ1TEyMb2jr2ItkaBRImuuko4Ksc1u5APiLncOmUJm+2
+KanShH9fNMKOs8w=
+-----END CERTIFICATE-----
+`
+const tlsKey = 
+`-----BEGIN EC PARAMETERS-----
+BgUrgQQAIg==
+-----END EC PARAMETERS-----
+-----BEGIN EC PRIVATE KEY-----
+MIGkAgEBBDCtJtoa5ezMGl5vOm7F2wJfBnceiLIapTZEZcdMBdp3nuWXyZ3ENbIF
+wbbvIrioZnagBwYFK4EEACKhZANiAAQYlwM8MU+efdXMGIBMRuVkzuC5DbmvX3sW
+efebB4KE7Ppdd4ZZicXRQ2lWmrH+1kvwPvIUDvAixLwB1TT4V4khO8kMi0XO5GTc
+XSl1RR/+kchK1i99l8A/5Diobc6/Uwo=
+-----END EC PRIVATE KEY-----
+`
 
 var _ = Describe("server", func() {
 	testing.WithConfig(func(cfg **config.Config) {
+		var tlsCertificateFile string
+		var tlsCertificateKeyFile string
 		var temp_asset_dir *testing.TmpDirectory
+
 		BeforeSuite(func() {
```

Comment:
We should probably split gzip and tls tests, maybe even put them in different files.

Also, I think we want to switch to `BeforeEach` instead of `BeforeSuite` because tls certificates part is only relevant for tls tests and gzip files part is only relevant for gzip tests.